src/queries.ts

```diff
@@ -47,43 +42,93 @@ async function getStakedPSTHolders(): Promise<StakedPSTHolders> {
 	return Object.fromEntries(stakedForAtLeastOneMonth);
 }
 
+/**
+ * Queries GQL for all ArDrive transactions within a range of blocks
+ * @param minBlock an integer representing the block from where to query the data
+ * @param maxBlock an integer representing the block until where to query the data
+ * @returns {Promise<GQLEdgeInterface[]>} the edges of the GQL result
+ */
 export async function getAllTransactionsWithin(minBlock: number, maxBlock: number): Promise<GQLEdgeInterface[]> {
 	const allEdges: GQLEdgeInterface[] = [];
+
+	const blockHeight = await getBlockHeight();
+	const trustedHeight = blockHeight - 50;
+
 	let hasNextPage = true;
+	let prevBlock = minBlock - 1;
 
 	while (hasNextPage) {
-		const query = createQuery(minBlock, maxBlock);
+		const query = createQuery(prevBlock + 1, Math.min(maxBlock, trustedHeight));
 		const response = await sendQuery(query);
-		allEdges.push(...response.edges);
-		hasNextPage = response.pageInfo.hasNextPage;
+		if (response.edges.length) {
+			console.log(`Transactions count: ${response.edges.length}`);
```

Comment:
Is this logging from debugging or are we going to keep some info for production?

We would probably see the output on the GitHub Action if thats what we ultimately use

Reply:
We'll have to tell redstone to write to STDERR as well